lib/src/nt4_client.dart

```diff
@@ -332,26 +365,52 @@ class NT4Client {
   }
 
   void _wsSendBinary(dynamic data) {
-    _ws?.sink.add(data);
+    _mainWebsocket?.sink.add(data);
   }
 
   void _wsConnect() async {
+    if (_serverConnectionActive) {
+      return;
+    }
+
     _clientId = Random().nextInt(99999999);
 
     String serverAddr = 'ws://$serverBaseAddress:5810/nt/DartClient_$_clientId';
 
-    _ws = WebSocketChannel.connect(Uri.parse(serverAddr),
-        protocols: ['networktables.first.wpi.edu']);
+    _mainWebsocket =
+        WebSocketChannel.connect(Uri.parse(serverAddr), protocols: [
+      'networktables.first.wpi.edu',
+      'v4.1.networktables.first.wpi.edu',
+    ]);
```

Comment:
How does this handle both protocols being supported? Does the NT server see that it supports both and chooses 4.1?

Reply:
NT will choose the protocol that is preferred, which for WPILib 2024 would be 4.1. If it were connecting to a pre-2024 WPILib NT server, it would prefer using the normal one since that's the only protocol it supports. This implementation was heavily inspired from [AdvantageScope's NT client](https://github.com/Mechanical-Advantage/AdvantageScope/blob/b7105a80aa02b2f4837c9555386ef952a5a44c1a/src/hub/dataSources/NT4.ts#L659)